include/algorithms/synthesis/syrec_synthesis.hpp

```diff
@@ -44,115 +34,102 @@ namespace syrec::internal {
 namespace syrec {
     using namespace internal;
 
-    class standard_syrec_synthesizer {
+    class SyrecSynthesis {
     public:
         std::stack<unsigned>               exp_opp;
-        std::stack<std::vector<unsigned>>  exp_lhss, exp_rhss;
+        std::stack<std::vector<unsigned>>  exp_lhss;
+        std::stack<std::vector<unsigned>>  exp_rhss;
         bool                               sub_flag = false;
-        std::vector<unsigned>              op_vec, assign_op_vector, exp_op_vector;
-        std::vector<std::vector<unsigned>> exp_lhs_vector, exp_rhs_vector;
+        std::vector<unsigned>              op_vec;
+        std::vector<unsigned>              assign_op_vector;
+        std::vector<unsigned>              exp_op_vector;
+        std::vector<std::vector<unsigned>> exp_lhs_vector;
+        std::vector<std::vector<unsigned>> exp_rhs_vector;
+
+        using var_lines_map = std::map<variable::ptr, unsigned int>;
 
-        typedef std::map<variable::ptr, unsigned> var_lines_map;
+        explicit SyrecSynthesis(circuit& circ);
+        virtual ~SyrecSynthesis() = default;
 
-        standard_syrec_synthesizer(circuit& circ, const program& prog);
+        void add_variables(circuit& circ, const variable::vec& variables);
+        void set_main_module(const module::ptr& main_module);
 
-        virtual ~standard_syrec_synthesizer() = default;
+    protected:
+        // TODO: please find a proper name for this function, this is just a placeholder for now
```

Comment:
I left this as a TODO since I did not have the time to think of a proper name and just noticed that this fragment of code is the only thing that differs in both synthesisers for many functions.

Reply:
Since its processing the main statement, I suggest we keep the same name.